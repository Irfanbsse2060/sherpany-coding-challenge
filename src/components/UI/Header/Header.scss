@import "../../../styles/variables";

.header {
  overflow: hidden;
  background-color: #333;
  position: sticky;
  position: -webkit-sticky;
  top: 0;
  margin: 0;
  list-style: none;
  padding: 0;

  &__item {
    a {
      float: left;
      display: block;
      color: white;
      text-align: center;
      padding: 14px 20px;
      text-decoration: none;
    }
    a:hover {
      background-color: #ddd;
      color: black;
    }

    a.active {
      background-color: #666;
      color: white;
    }

    @media screen and (max-width: 400px) {
       a {
        float: none;
        width: 100%;
      }
    }

  }
}

@media screen and (max-width: 400px) {
  .header a {
    float: none;
    width: 100%;
  }
}
