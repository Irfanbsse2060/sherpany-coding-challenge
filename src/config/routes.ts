export const routePaths = {
    home: "/",
    settings: "/settings"
}


export const linkWithRoutes = [
    {
        label: "Home",
        url: routePaths.home
    },
    {
        label: "Settings",
        url: routePaths.settings
    }
]
