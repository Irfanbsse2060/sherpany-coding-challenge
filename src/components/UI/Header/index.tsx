import React from 'react'
import { Link } from 'react-router-dom'
import './Header.scss'

interface HeaderLinks {
    label: string
    url: string
}

interface HeaderProps {
    linkList: HeaderLinks[]
}


export default function Header(props: HeaderProps) {
    const {linkList} = props
    return (
        <ul data-testid={'header'} className='header'>
            {
                linkList.map((item) => {
                    return (
                        <li key={item.url} className='header__item'>
                            <Link to={item.url}>{item.label}</Link>
                        </li>

                    )

                })
            }
        </ul>
    )

}
