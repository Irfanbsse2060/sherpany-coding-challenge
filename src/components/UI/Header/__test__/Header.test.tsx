import React from 'react';
import {render} from '@testing-library/react';
import Header from '../index'
import {BrowserRouter as Router} from "react-router-dom";
import {linkWithRoutes} from "../../../../config/routes";

describe('Header', () => {
    test('should be able to render', () => {
        const {container} = render(<Router>
            <Header linkList={linkWithRoutes}/>
        </Router>);
        expect(container).toMatchSnapshot()
    })


})
