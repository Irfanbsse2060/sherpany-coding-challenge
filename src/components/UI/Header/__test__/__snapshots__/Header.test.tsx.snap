// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should be able to render 1`] = `
<div>
  <ul
    class="header"
    data-testid="header"
  >
    <li
      class="header__item"
    >
      <a
        href="/"
      >
        Home
      </a>
    </li>
    <li
      class="header__item"
    >
      <a
        href="/settings"
      >
        Settings
      </a>
    </li>
  </ul>
</div>
`;
